docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/gpu-device/remote-tensor-api-gpu-plugin.rst

```diff
@@ -350,7 +350,139 @@ For more details, see the code snippets below:
+                           .set_memory_type(ov::intel_gpu::memory_type::surface);
+         p.input().preprocess().convert_color(ov::preprocess::ColorFormat::BGR);
+         p.input().model().set_layout("NCHW");
+         model = p.build();
+
+         CComPtr<ID3D11Device> device_ptr = get_d3d_device_ptr()
+         // create the shared context object
+         auto shared_va_context = ov::intel_gpu::ocl::D3DContext(core, device_ptr);
+         // compile model within a shared context
+         auto compiled_model = core.compile_model(model, shared_va_context);
+
+         auto input0 = model->get_parameters().at(0);
+         auto input1 = model->get_parameters().at(1);
+
+         auto shape = input0->get_shape();
+         auto width = shape[1];
+         auto height = shape[2];
+
+         D3D11_TEXTURE2D_DESC texture_description = get_texture_desc();
+         CComPtr<ID3D11Texture2D> dx11_texture = get_texture();
+         //     ...
+         //wrap decoder output into RemoteBlobs and set it as inference input
+         auto nv12_blob = shared_va_context.create_tensor_nv12(texture_description.Heights, texture_description.Width, dx11_texture);
+
+         auto infer_request = compiled_model.create_infer_request();
+         infer_request.set_tensor(input0->get_friendly_name(), nv12_blob.first);
+         infer_request.set_tensor(input1->get_friendly_name(), nv12_blob.second);
+         infer_request.start_async();
+         infer_request.wait();
+
+   .. tab-item:: VASurfaceID
+      :sync: vasurfaceid
+
+      .. code-block:: cpp
+
+         using namespace ov::preprocess;
+         auto p = PrePostProcessor(model);
+         p.input().tensor().set_element_type(ov::element::u8)
+                           .set_color_format(ov::preprocess::ColorFormat::NV12_TWO_PLANES, {"y", "uv"})
+                           .set_memory_type(ov::intel_gpu::memory_type::surface);
+         p.input().preprocess().convert_color(ov::preprocess::ColorFormat::BGR);
+         p.input().model().set_layout("NCHW");
+         model = p.build();
+
+         CComPtr<ID3D11Device> device_ptr = get_d3d_device_ptr()
+         // create the shared context object
+         auto shared_va_context = ov::intel_gpu::ocl::VAContext(core, device_ptr);
+         // compile model within a shared context
+         auto compiled_model = core.compile_model(model, shared_va_context);
+
+         auto input0 = model->get_parameters().at(0);
+         auto input1 = model->get_parameters().at(1);
+
+         auto shape = input0->get_shape();
+         auto width = shape[1];
+         auto height = shape[2];
+
+         D3D11_TEXTURE2D_DESC texture_description = get_texture_desc();
+         CComPtr<ID3D11Texture2D> dx11_texture = get_texture();
+         //     ...
+         //wrap decoder output into RemoteBlobs and set it as inference input
+         auto nv12_blob = shared_va_context.create_tensor_nv12(texture_description.Heights, texture_description.Width, dx11_texture);
```

Comment:
That's not correct for VA. Please extract code snippet from `Direct Consuming of the NV12 VAAPI Video Decoder Surface on Linux` as is and put it here